datafusion/expr/src/binary_rule.rs

```diff
@@ -516,6 +516,8 @@ fn temporal_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataTyp
     use arrow::datatypes::DataType::*;
     use arrow::datatypes::TimeUnit;
     match (lhs_type, rhs_type) {
+        (Date64, Date32) => Some(Date64),
```

Comment:
🤔  I think coercing from Date32 -> Date64 is probably fine, but going the other way loses precision, right?

Reply:
My understanding of this code is that it is finding the common type that both lhs_type and rhs_type can safely be cast to, so any combination of `Date32` and `Date64` can be cast to `Date64`

Reply:
Yes, you are correct. Sorry about that